remote_fetch.go

```diff
@@ -92,16 +92,34 @@ func decompressResponse(bodyBytes []byte, contentLength int64, cacheLocator Cach
 				return errorExtractingPostgres(err)
 			}
 
-			cacheLocation, _ := cacheLocator()
+			// to prevent file corruption when multiple processes attempt to download at the same time we download
+			// first to a cache location, and then move the file into place.
+			tmp, err := os.CreateTemp("", "embedded_postgres")
+			if err != nil {
+				return errorExtractingPostgres(err)
+			}
 
-			if err := os.MkdirAll(filepath.Dir(cacheLocation), 0755); err != nil {
+			if err := os.WriteFile(tmp.Name(), archiveBytes, file.FileHeader.Mode()); err != nil {
 				return errorExtractingPostgres(err)
 			}
 
-			if err := os.WriteFile(cacheLocation, archiveBytes, file.FileHeader.Mode()); err != nil {
+			cacheLocation, _ := cacheLocator()
+			if err := os.MkdirAll(filepath.Dir(cacheLocation), 0755); err != nil {
 				return errorExtractingPostgres(err)
 			}
 
+			if _, err := os.Stat(cacheLocation); errors.Is(err, os.ErrNotExist) {
+				if err := os.WriteFile(cacheLocation, archiveBytes, file.FileHeader.Mode()); err != nil {
```

Comment:
This has a race. Two processes can see the file doesn't exist and then try to write it. To work around Windows' rename, you need to try the rename and ignore the error if it is "file already exists".